aspnet/fundamentals/logging/sample/src/LoggingSample/RequestLoggerMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Internal;
+
+namespace LoggingSample
+{
+    public class RequestLoggerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly static Random _random = new Random();
+
+        public RequestLoggerMiddleware(RequestDelegate next, 
+            ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<RequestLoggerMiddleware>();
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var eventId = _random.Next();
+            _logger.LogInformation(eventId, 
+                "{eventId}: Handling request: {path}", 
+                eventId, context.Request.Path);
+            try
+            {
+                await _next.Invoke(context);
+            }
+            catch (Exception ex)
+            {
+                if (_logger.IsEnabled(LogLevel.Critical))
+                {
+                    // perform expensive operation related to logging
+
+                    var values = new FormattedLogValues("{eventId}: Unexpected error handling request {path}:",
```

Comment:
I was looking for an overload for LogCritical that would allow me to pass in the same parameters I'm passing into LogInformation (eventId, message) as well as an exception. I found (eventId, state, exception) but it looks like I could use (eventId, message, exception).

Updated. Should FormattedLogValues be internal or otherwise restricted from non-framework use?